core/src/main/scala/org/http4s/rho/AuthedContext.scala

```diff
@@ -47,22 +49,23 @@ class AuthedContext[U] {
     * @param service [[HttpService]] to convert
     * @return An `AuthedService` which can be mounted by http4s servers.
     */
-  def toService(service: HttpService): AuthedService[U] = {
-    Service.lift { case AuthedRequest(authInfo, req) =>
-      service(req.withAttribute[U](authKey, authInfo))
+  def toService(service: HttpService[F]): AuthedService[U, F] = {
+    type O[A] = OptionT[F, A]
+
+    Kleisli[O, AuthedRequest[F, U], Response[F]] { (a: AuthedRequest[F, U]) =>
+      service(a.req.withAttribute[U](authKey, a.authInfo))
     }
   }
 
   /* Get the authInfo object from request. */
-  def getAuth(req: Request): U = {
+  def getAuth(req: Request[F]): U = {
     req.attributes.get[U](authKey).get
   }
 
-  final val auth: TypedHeader[U :: HNil] = rho.genericRequestHeaderCapture { req =>
+  def auth(): TypedHeader[F, U :: HNil] = RhoDsl[F].genericRequestHeaderCapture[U] { req =>
```

Comment:
Does this really need a `()`?

Reply:
You can still access it without the parens.